docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Recommendation/OneClassMatrixFactorizationWithOptions.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Samples.Dynamic.Trainers.Recommendation
+{
+    public static class OneClassMatrixFactorizationWithOptions
+    {
+        // This example shows the use of ML.NET's one-class matrix factorization module which implements
+        // Algorithm 1 in a <a href="https://www.csie.ntu.edu.tw/~cjlin/papers/one-class-mf/biased-mf-sdm-with-supp.pdf">paper</a>.
+        // See page 28 in of <a href="https://www.csie.ntu.edu.tw/~cjlin/talks/facebook.pdf">slides</a> for a brief introduction to
+        // one-class matrix factorization.
+        // In this example we will create in-memory data and then use it to train a one-class matrix factorization model.
+        // Afterward, prediction values are reported.
+        // To run this example, it requires installation of additional nuget package
+        // <a href="https://www.nuget.org/packages/Microsoft.ML.Recommender/">Microsoft.ML.Recommender</a>.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging,
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 0);
+
+            // Get a small in-memory dataset.
+            GetOneClassMatrix(out List<MatrixElement> data, out List<MatrixElement> testData);
+
+            // Convert the in-memory matrix into an IDataView so that ML.NET components can consume it.
+            var dataView = mlContext.Data.LoadFromEnumerable(data);
+
+            // Create a matrix factorization trainer which may consume "Value" as the training label, "MatrixColumnIndex" as the
```

Comment:
>may consume [](start = 59, length = 11)

takes

Reply:
Thanks.

---
In reply to: [274575597](https://github.com/dotnet/machinelearning/pull/3282#discussion_r274575597) [](ancestors = 274575597)